DigitalLearningSolutions.Data/Services/ActivityService.cs

```diff
@@ -0,0 +1,64 @@
+﻿namespace DigitalLearningSolutions.Data.Services
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using DigitalLearningSolutions.Data.DataServices;
+    using DigitalLearningSolutions.Data.Models.TrackingSystem;
+
+    public interface IActivityService
+    {
+        public IEnumerable<MonthOfActivity> GetRecentActivity(int centreId);
+    }
+
+    public class ActivityService : IActivityService
+    {
+        private readonly IActivityDataService activityDataService;
+        private readonly IClockService clockService;
+
+        public ActivityService(IActivityDataService activityDataService, IClockService clockService)
+        {
+            this.activityDataService = activityDataService;
+            this.clockService = clockService;
+        }
+
+        public IEnumerable<MonthOfActivity> GetRecentActivity(int centreId)
+        {
+            var endTime = clockService.UtcNow;
+            var startTime = endTime.AddYears(-1);
+
+            var activityData = activityDataService.GetActivityInRangeByMonth(centreId, startTime, endTime).ToList();
+
+            var months = GenerateBlankMonthsBetweenDates(startTime, endTime).ToList();
+            months.Reverse();
+
+            foreach (var month in months)
+            {
+                var monthData =
+                    activityData.SingleOrDefault(data => data.Year == month.Year && data.Month == month.Month);
+                if (monthData != null)
+                {
+                    month.Completions = monthData.Completions;
+                    month.Registrations = monthData.Registrations;
+                    month.Evaluations = monthData.Evaluations;
+                }
+            }
+
+            return months;
+        }
+
+        private IEnumerable<MonthOfActivity> GenerateBlankMonthsBetweenDates(DateTime startDate, DateTime endDate)

```

Comment:
I think this is great, but I'd probably return a `IEnumerable<(int year, int month)>` from this method and pull it into a helper, where we can write very specific unit tests for this and reuse it in the future.

We could also do something like this above (a simple map to the object we need):
```
var monthSlots = DateTimeHelper.GenerateMonthSlots(..);
return monthSlots.Select(slot => {
  var monthlyUsageStats = ...
  return new MonthOfActivity(slot, monthlyUsageStats ) // Need to add a constructor
});
```